packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/LinkToCategoriesTests.java

```diff
@@ -0,0 +1,350 @@
+        STEP("Verify if all three categories are present in file metadata");
+        final RestNodeModel fileNode = restClient.authenticateUser(user).withCoreAPI().usingNode(file).getNode();
+
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains("cm:categories");
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(category.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(secondCategory.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(thirdCategory.getId());
+    }
+
+    /**
+     * Try to link content to category as a user without read permission and expect 403 (Forbidden)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_asUserWithoutReadPermissionAndExpect403()
+    {
+        STEP("Try to link content to a category using user without read permission and expect 403");
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(category.getId());
+        final UserModel userWithoutRights = dataUser.createRandomTestUser();
+        restClient.authenticateUser(userWithoutRights).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(FORBIDDEN);
+    }
+
+    /**
+     * Try to link content to category as a user without change and expect 403 (Forbidden)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_asUserWithoutChangePermissionAndExpect403()
+    {
+        STEP("Create another user as a consumer for file");
+        final UserModel consumer = dataUser.createRandomTestUser();
+        addPermissionsForUser(consumer.getUsername(), "Consumer", file);
+
+        STEP("Try to link content to a category using user without change permission and expect 403");
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(category.getId());
+        restClient.authenticateUser(consumer).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(FORBIDDEN);
+    }
+
+    /**
+     * Try to link content to category as a owner and expect 403 (Forbidden)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_asOwnerAndExpect403()
+    {
+        STEP("Use admin to create a private site");
+        final SiteModel privateSite = dataSite.usingAdmin().createPrivateRandomSite();
+
+        STEP("Add the user to the site, let him create a folder and then evict him from the site again");
+        dataUser.addUserToSite(user, privateSite, SiteManager);
+        final FolderModel privateFolder = dataContent.usingUser(user).usingSite(privateSite).createFolder();
+        final FileModel privateFile = dataContent.usingUser(user).usingResource(privateFolder).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        dataUser.removeUserFromSite(user, privateSite);
+
+        STEP("Try to link content to a category as owner and expect 403");
```

Comment:
I don't understand this. The owner should be able to link the content to a category.